tree_clipper/ui.py

```diff
@@ -190,32 +251,98 @@ class SCENE_OT_Tree_Clipper_Import_Cache(bpy.types.Operator):
     allow_version_mismatch: bpy.props.BoolProperty(name="Ignore Version", default=False)  # type: ignore
     debug_prints: bpy.props.BoolProperty(name="Debug on Console", default=False)  # type: ignore
 
-    def invoke(self, context, _):
+    def invoke(
+        self, context: bpy.types.Context, event: bpy.types.Event
+    ) -> set["rna_enums.OperatorReturnItems"]:
+        assert isinstance(_INTERMEDIATE_IMPORT_CACHE, ImportIntermediate)
+        assert hasattr(context.scene, "tree_clipper_external_import_items")
+        assert isinstance(
+            context.scene.tree_clipper_external_import_items,
+            Tree_Clipper_External_Import_Items,
+        )
+        context.scene.tree_clipper_external_import_items.items.clear()
+        for (
+            external_id,
+            external_item,
+        ) in _INTERMEDIATE_IMPORT_CACHE.get_external().items():
+            if external_item["skip"]:
+                continue
+            item = context.scene.tree_clipper_external_import_items.items.add()
+            item.external_id = int(external_id)
+            item.description = external_item["description"]
+            item.set_active_pointer_type(external_item["fixed_type_name"])
+
         return context.window_manager.invoke_props_dialog(self)
 
-    def execute(self, _context):
+    def execute(
+        self, context: bpy.types.Context
+    ) -> set["rna_enums.OperatorReturnItems"]:
         global _INTERMEDIATE_IMPORT_CACHE
         assert isinstance(_INTERMEDIATE_IMPORT_CACHE, ImportIntermediate)
+        assert hasattr(context.scene, "tree_clipper_external_import_items")
+        assert isinstance(
+            context.scene.tree_clipper_external_import_items,
+            Tree_Clipper_External_Import_Items,
+        )
+
+        # collect what is set from the UI
+        getters: dict[int, GETTER] = dict(
```

Comment:
I think this would be better suited as a function that takes items that is called here, for potential re-use & more robust testing